plugin/pkg/scheduler/factory/factory.go

```diff
@@ -1066,12 +1159,22 @@ func (factory *configFactory) MakeDefaultErrorFunc(backoff *util.PodBackoff, pod
 				if err == nil {
 					if len(pod.Spec.NodeName) == 0 {
 						podQueue.AddUnschedulableIfNotPresent(pod)
+					} else {
```

Comment:
Please remove this one too.

Reply:
What about the scenario where the pod is updated/deleted while it is being processed in `scheduleOne`?  The two checks here were supposed to be for that case.

Reply:
The pod is unassinged while it is being processed in `scheduleOne`. So, if it is updated or deleted, we will get the corresponding event for the unassigned pod.

Reply:
What if:
1. Start processing pod in scheduleOne
2. Pod gets deleted, unassigned event comes and clears cache
3. The processing in scheduleOne updates the cache with the pod
4. Assume/bind fails because pod doesn't exist anymore
5. Cache needs to be cleaned up somehow

Reply:
The scenario is valid, but in that case pod will not be assigned (len(pod.Spec.NodeName) == 0) and the pod will not be found, so err != nil. So, the check you have added below to `if errors.IsNotFound(err) {` should suffice.

Reply:
What if it was an update event that caused the pod to be assigned?

Reply:
The way our event handlers are set up is that they have a filter that check for assignment of pods. So, whenever a pod is assigned, we received one "delete" event for the unassigned pod followed by an "add" event for the assigned pod. In other words, an unassigned pod won't become assigned in an update.

Reply:
Had an offline discussion with @msau42. This is correct.